packages/web/src/common/components/Filter/index.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useRef, useState } from "react";
+
+import styled from "styled-components";
+
+import FilterButton from "./_atomic/FilterButton";
+import FilterDropdown, { FilterProps } from "./_atomic/FilterDropdown";
+
+const FilterWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-end;
+  gap: 4px;
+`;
+
+const Filter: React.FC<FilterProps> = ({
+  itemList,
+  selectedList,
+  setSelectedList,
+}) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const wrapperRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    function handleClickOutside(event: MouseEvent) {
+      if (
+        wrapperRef.current &&
+        !wrapperRef.current.contains(event.target as Node)
+      ) {
+        setIsOpen(false);
+      }
+    }
+
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, []);
+
+  return (
+    <div ref={wrapperRef} style={{ display: "inline-flex" }}>
+      <FilterWrapper>
+        <FilterButton
+          isOpen={isOpen}
+          setIsOpen={setIsOpen}
+          itemList={itemList}
+          selectedList={selectedList}
+        />
+        {isOpen && (
+          <FilterDropdown
+            itemList={itemList}
+            selectedList={selectedList}
+            setSelectedList={setSelectedList}
+          />
+        )}
+      </FilterWrapper>
+    </div>
```

Comment:
오 이렇게 할거면 아예 아래처럼 하는건 어떨까
```suggestion
      <FilterWrapper>
      <div ref={wrapperRef}>
        <FilterButton
          isOpen={isOpen}
          setIsOpen={setIsOpen}
          itemList={itemList}
          selectedList={selectedList}
        />
        </div>
        {isOpen && (
          <FilterDropdown
            itemList={itemList}
            selectedList={selectedList}
            setSelectedList={setSelectedList}
          />
        )}
      </FilterWrapper>
```

Reply:
align-items: flex-end 이거랑 display: "inline-flex"를 둘다 넣는게 좀 애매함

Reply:
엇 근데 닫는 div 태그가 저기 있으면 작동을 안하지 않나요

Reply:
display: "inline-flex" --> div 크기 때문에 특정 부분 클릭해도 필터 안 닫히는 문제 수정
align-items: flex-end --> 옵션들이 오른쪽 정렬되어서 표현되도록 
요런 의도에서 쓴 거긴 한데 더 좋은 방법이 있는지 고민해보겠습니당

Reply:
오 그런 의도였다면 좋아요 
div 그대로 쓰지 말고 styled component로 wrapper 하나 정의해서 사용하도록 수정만 해주세요!